acp/ideas_module.php

```diff
@@ -144,7 +144,16 @@ public function main($id, $mode)
 		}
 
 		// Check if Ideas forum is selected and apply relevant settings if it is
-		if ($submit_forum_setup)
+		// But display the confirm box first
+		if ($submit_forum_setup && !confirm_box(true))
```

Comment:
Can you just wrap this all up inside the `if ($submit_forum_setup)` like normal?
```php
<?php

if ($submit_forum_setup)
{
	if (confirm_box(true))
	{
		// do all the work
	}
	else
	{
		confirm_box(false, $this->language->lang('ACP_IDEAS_FORUM_SETUP_CONFIRM'), build_hidden_fields(array(
			'i'			=> $id,
			'mode'		=> $mode,
			'ideas_forum_setup'	=> $submit_forum_setup,
		)));
	}
}
```